### PR TITLE
packaging/fedora: Ensure vendor/ is empty for builds and fix spec to build current master

### PR DESCRIPTION
**Pull request by Conan-Kudo**
### DO NOT SQUASH MERGE THIS ###

For Fedora, we're building with unbundled components, but it seems that vendored stuff is leaking into spread tests and other things.

So let's just enforce that we build the right way so that we don't get misled by spread tests.

In addition, some basic fixes to the spec file so that it matches reality have been done, too.

Unfortunately, because CI checkmarks rule over everything and this *needs* to be in 2.28 (because the build is broken), a temporary hack has been pushed to fix the build too.

Signed-off-by: Neal Gompa <ngompa13@gmail.com>

### CODE REVIEW

**Comment by Conan-Kudo** [comment]
Note that I *expect* spread to fail at this point, and I need it to fail because I know the spec is outdated vs what's changed.

File: packaging/fedora/snapd.spec [hunk 1]
```diff
@@ -98,6 +99,7 @@ BuildRequires: golang(golang.org/x/crypto/ssh/terminal)
 BuildRequires: golang(golang.org/x/net/context)
 BuildRequires: golang(golang.org/x/net/context/ctxhttp)
 BuildRequires: golang(gopkg.in/check.v1)
+BuildRequires: golang(gopkg.in/cheggaaa/pb.v1)
```
Comment (zyga) [inline packaging/fedora/snapd.spec]: We could depend on it and then do some symlinks at the right spot to unblock the build and land it for now.

Comment (Conan-Kudo) [reply packaging/fedora/snapd.spec]: Well, this `golang()` name doesn't exist yet at all...

Comment (Conan-Kudo) [reply packaging/fedora/snapd.spec]: I implemented this, but I'm extremely unhappy that this situation even happened to begin with.

**Comment by Conan-Kudo** [comment]
@mvo5 @zyga One of you please merge this, ASAP!

@mvo5 This needs to be folded back into the 2.28 release branch as well.

**Pull request merged**